src/databricks/labs/ucx/install.py

```diff
@@ -233,58 +272,43 @@ def _apply_upgrades(self):
             logger.warning(f"Installed version is too old: {err}")
             return
 
-    def _configure_new_installation(self) -> WorkspaceConfig:
-        logger.info("Please answer a couple of questions to configure Unity Catalog migration")
-        HiveMetastoreLineageEnabler(self._ws).apply(self._prompts)
-        inventory_database = self._prompts.question(
-            "Inventory Database stored in hive_metastore", default="ucx", valid_regex=r"^\w+$"
-        )
-
-        self._check_inventory_database_exists(inventory_database)
+    def _configure_new_installation(self, default_config: WorkspaceConfig | None = None) -> WorkspaceConfig:
+        if default_config is None:
+            default_config = self._prompt_for_new_installation()
+        HiveMetastoreLineageEnabler(self._ws).apply(self._prompts, self._is_account_install)
+        self._check_inventory_database_exists(default_config.inventory_database)
         warehouse_id = self._configure_warehouse()
-        configure_groups = ConfigureGroups(self._prompts)
-        configure_groups.run()
-        log_level = self._prompts.question("Log level", default="INFO").upper()
-        num_threads = int(self._prompts.question("Number of threads", default="8", valid_number=True))
 
         policy_id, instance_profile, spark_conf_dict, instance_pool_id = self._policy_installer.create(
-            inventory_database
+            default_config.inventory_database
         )
 
         # Save configurable values for table migration cluster
         min_workers, max_workers, spark_conf_dict = self._config_table_migration(spark_conf_dict)
 
-        # Check if terraform is being used
-        is_terraform_used = self._prompts.confirm("Do you use Terraform to deploy your infrastructure?")
-
-        config = WorkspaceConfig(
-            inventory_database=inventory_database,
-            workspace_group_regex=configure_groups.workspace_group_regex,
-            workspace_group_replace=configure_groups.workspace_group_replace,
-            account_group_regex=configure_groups.account_group_regex,
-            group_match_by_external_id=configure_groups.group_match_by_external_id,  # type: ignore[arg-type]
-            include_group_names=configure_groups.include_group_names,
-            renamed_group_prefix=configure_groups.renamed_group_prefix,
+        config = dataclasses.replace(
+            default_config,
             warehouse_id=warehouse_id,
-            log_level=log_level,
-            num_threads=num_threads,
             instance_profile=instance_profile,
             spark_conf=spark_conf_dict,
             min_workers=min_workers,
             max_workers=max_workers,
             policy_id=policy_id,
             instance_pool_id=instance_pool_id,
-            is_terraform_used=is_terraform_used,
-            include_databases=self._select_databases(),
+            installed_workspace_ids=self._installed_workspace_ids,
```

Comment:
Shouldn't we add a list of workspaces only after we successfully installed on a particular workspace? That way it's clear that operation succeeded or not, preventing downstream bugs